tests/templates/test_subroutines/test_trotter.py

```diff
@@ -162,6 +163,47 @@
 }
 
 
+def _generate_simple_decomp(coeffs, ops, time, order, n):
+    """Given coeffs, ops and a time argument in a given framework, generate the
+    Trotter product for order and number of trotter steps."""
+    decomp = []
+    if order == 1:
+        decomp.extend(qml.exp(op, coeff * (time / n) * 1j) for coeff, op in zip(coeffs, ops))
+
+    coeffs_ops = zip(coeffs, ops)
+
+    if get_interface(coeffs) == "torch":
+        import torch
+
+        coeffs_ops_reversed = zip(torch.flip(coeffs, dims=(0,)), ops[-1::-1])
+    else:
+        coeffs_ops_reversed = zip(coeffs[-1::-1], ops[-1::-1])
+
+    if order == 2:
+        decomp.extend(qml.exp(op, coeff * (time / n) * 1j / 2) for coeff, op in coeffs_ops)
+        decomp.extend(qml.exp(op, coeff * (time / n) * 1j / 2) for coeff, op in coeffs_ops_reversed)
+
+    if order == 4:
```

Comment:
Do we use this branch in any tests? (fine to keep it even if we do not use it, but we need to make sure it is accurate)

Reply:
We have tested order 4 for the decomposition, so no need to have it for gradients.

Reply:
Added 4 decomposition to tests 👍🏼